sklearn/pipeline.py

```diff
@@ -221,22 +227,58 @@ def _validate_steps(self):
                             "'%s' (type %s) doesn't"
                             % (estimator, type(estimator)))
 
+        return deepcopy(self._steps)
```

Comment:
I'm not sure a deepcopy is appropriate. Shouldn't we be doing `[(name, clone(est)) for name, est in self._steps]`?